src/core/libraries/rtc/rtc.cpp

```diff
@@ -66,64 +154,211 @@ int PS4_SYSV_ABI sceRtcFormatRFC3339PreciseLocalTime() {
     return ORBIS_OK;
 }
 
-int PS4_SYSV_ABI sceRtcGetCurrentAdNetworkTick() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcGetCurrentAdNetworkTick(OrbisRtcTick* pTick) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTick == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    Kernel::OrbisKernelTimespec clocktime;
+    int returnValue = Kernel::sceKernelClockGettime(Kernel::ORBIS_CLOCK_EXT_NETWORK, &clocktime);
+
+    if (returnValue == 0) {
+        pTick->tick = clocktime.tv_nsec / 1000 + clocktime.tv_sec * 1000000 + 0xdcbffeff2bc000;
+    } else {
+        return ORBIS_RTC_ERROR_NOT_INITIALIZED;
+    }
+
+    return 0;
 }
 
-int PS4_SYSV_ABI sceRtcGetCurrentClock() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcGetCurrentClock(OrbisRtcDateTime* pTime, int minutesToAdd) {
```

Comment:
```suggestion
int PS4_SYSV_ABI sceRtcGetCurrentClock(OrbisRtcDateTime* pTime, int timeZone) {
```
This second parameter is time zone type, not minutes, for example UTC is `timeZone = 0` type. Maybe you have seen some indication otherwise?

Reply:
I named it that because it ends up getting passed to sceRtcTickAddMinutes as param 3, which I have been assuming was the minutes being added, could you take a look at the params on that function? maybe i misunderstood how it works

Reply:
This third parameter is `int64_t lAdd`, `long long int`, where are the usage which places this timeZone into the `sceRtcAddMinutes`?